tasks/sep/base-001-MCP-L1/VALIDATION.md

```diff
@@ -0,0 +1,341 @@
+  **Before** address matches both of the following cast calls (please consider changing out the rpc
+  url):
+  1. what is returned by calling `AddressManager.getAddress()`:
+   ```
+   cast call 0x709c2B8ef4A9feFc629A8a2C1AF424Dc5BD6ad1B 'getAddress(string)(address)' 'OVM_L1CrossDomainMessenger' --rpc-url https://ethereum-sepolia.publicnode.com
+   ```
+  2. what is currently stored at the key:
+   ```
+   cast storage 0x709c2B8ef4A9feFc629A8a2C1AF424Dc5BD6ad1B 0x515216935740e67dfdda5cf8e248ea32b3277787818ab59153061ac875c9385e --rpc-url https://ethereum-sepolia.publicnode.com
+   ```
+
+### `0x84457ca9d0163fbc4bbfe4dfbb20ba46e48df254` (`L2OutputOracleProxy`)
+
+todo: there are 7 state changes in the simulated tx vs the templates 6
+
+Links:
+- [Etherscan](https://sepolia.etherscan.io/address/0x84457ca9d0163fbc4bbfe4dfbb20ba46e48df254)
+- [Superchain Registry](https://github.com/ethereum-optimism/superchain-registry/blob/a62fd38c9239459b27ea50a2df895e89a3305382/superchain/extra/addresses/sepolia/base.json#L6)
+
+State Changes:
+todo: new state change, meaning needs to be checked
+- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000000` <br/>
+  **Before:** `0x0000000000000000000000000000000000000000000000000000000000000002` <br/>
+  **After:** `0x0000000000000000000000000000000000000000000000000000000000000001` <br/>
+  **Meaning:** The `initialized` boolean is set to `true`. The correctness of
+   this slot is attested to in the Optimism repo at [storageLayout/L2OutputOracle.json](https://github.com/ethereum-optimism/optimism/blob/e6ef3a900c42c8722e72c2e2314027f85d12ced5/packages/contracts-bedrock/snapshots/storageLayout/L2OutputOracle.json#L2-L15). This state diff will only appear in contracts that were previously not initializable. Other contracts may be reinitialized but it does not show in the state diff because the storage diff is a noop.
+
+- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000004` <br/>
+  **Before:** `0x000000000000000000000000da3037ff70ac92cd867c683bd807e5a484857405` <br/>
+  **After:** `0x0000000000000000000000000000000000000000000000000000000000000078` <br/>
+  **Meaning:** Sets `submissionInterval` at slot `0x04` (4). The correctness of
+   this slot is attested to in the Optimism repo at [storageLayout/L2OutputOracle.json](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/snapshots/storageLayout/L2OutputOracle.json#L38-L42). `0x78` is 120 in decimal, which matches the current value found by `cast call 0x84457ca9d0163fbc4bbfe4dfbb20ba46e48df254 "SUBMISSION_INTERVAL()(uint256)" -r https://ethereum-sepolia.publicnode.com`.
+
+- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000005` <br/>
+  **Before:** `0x00000000000000000000000020044a0d104e9e788a0c984a2b7eae615afd046b` <br/>
```

Comment:
In other MCP upgrade, I believe these both of these `before` keys are zero? If so, let's take a closer look to understand what's being overwritten here and see if it's safe

Reply:
Based on the below slot changes, it looks like the value in slot 4 was the challenger address, and the value in slot 5 was proposer address. These are now in slots 6 and 7 respectively.

This change is surprising to me, since it implies we had a storage layout change?

Reply:
Here is Base Sepolia's [initially deployed commit](https://github.com/base-org/contract-deployments/blob/26783846e57316b02ed511d75edf4bf92dcc4269/sepolia/2023-09-26-deploy/.env#L1C11-L1C51) vs the [MCP contract release commit](https://github.com/ethereum-optimism/optimism/commit/e6ef3a900c42c8722e72c2e2314027f85d12ced5) [compared](https://github.com/ethereum-optimism/optimism/compare/a7ff5a811612fa338d0a6d6dd72dc2ec9badef6d..e6ef3a900c42c8722e72c2e2314027f85d12ced5#diff-f92c9308509097bf11b443e68883b1ed5a542f9d306988cdec0431831d9f1f5a).

If you look at the diff of the `L2OutputOracle.sol` and `L2OutputOracle.json`, it does look like the storage slots moved around and some new ones were added.

Reply:
Confirmed that there are storage layout changes here, but all moved variables are reset in `initialize`, so this change is okay.